flink-end-to-end-tests/flink-end-to-end-tests-kinesis-firehose/src/test/java/org/apache/flink/connector/firehose/table/test/KinesisFireHoseTableITTest.java

```diff
@@ -0,0 +1,236 @@
+                    .setNetwork(network)
+                    .setLogger(LOG)
+                    .dependsOn(mockFirehoseContainer)
+                    .build();
+
+    @Before
+    public void setup() throws Exception {
+        SdkAsyncHttpClient httpClient = createHttpClient(mockFirehoseContainer.getEndpoint());
+        s3AsyncClient = createS3Client(mockFirehoseContainer.getEndpoint(), httpClient);
+        firehoseAsyncClient = getFirehoseClient(mockFirehoseContainer.getEndpoint(), httpClient);
+        iamAsyncClient = createIamClient(mockFirehoseContainer.getEndpoint(), httpClient);
+        System.setProperty(SdkSystemSetting.CBOR_ENABLED.property(), "false");
+        LOG.info("1 - Creating the bucket for Firehose to deliver into...");
+        createBucket(s3AsyncClient, BUCKET_NAME);
+        LOG.info("2 - Creating the IAM Role for Firehose to write into the s3 bucket...");
+        createIAMRole(iamAsyncClient, ROLE_NAME);
+        LOG.info("3 - Creating the Firehose delivery stream...");
+        createDeliveryStream(STREAM_NAME, BUCKET_NAME, ROLE_ARN, firehoseAsyncClient);
+        LOG.info("Done setting up the localstack.");
+    }
+
+    @BeforeClass
+    public static void setupFlink() throws Exception {
+        FLINK.start();
+    }
+
+    @AfterClass
+    public static void stopFlink() {
+        FLINK.stop();
+    }
+
+    @After
+    public void teardown() {
+        System.clearProperty(SdkSystemSetting.CBOR_ENABLED.property());
+    }
+
+    @Test
+    public void testTableApiSink() throws Exception {
+        List<Order> orderList =
+                ImmutableList.of(
+                        new Order("A", 1),
+                        new Order("B", 2),
+                        new Order("C", 3),
+                        new Order("D", 4),
+                        new Order("E", 5));
+
+        executeSqlStatements(readSqlFile("send-orders.sql"));
+        List<Order> orders = readFromS3();
+        Assertions.assertThat(orders).containsAll(orderList);
+    }
+
+    private void executeSqlStatements(final List<String> sqlLines) throws Exception {
+        FLINK.submitSQLJob(
+                new SQLJobSubmission.SQLJobSubmissionBuilder(sqlLines)
+                        .addJars(sqlConnectorFirehoseJar)
+                        .build());
+    }
+
+    private List<String> readSqlFile(final String resourceName) throws Exception {
+        return Files.readAllLines(Paths.get(getClass().getResource("/" + resourceName).toURI()));
+    }
+
+    private List<Order> readFromS3() throws Exception {
+
+        Deadline deadline = Deadline.fromNow(Duration.ofMinutes(1));
+        List<S3Object> ordersObjects;
+        List<Order> orders = new ArrayList<>();
+        do {
+            Thread.sleep(1000);
+            ordersObjects = listBucketObjects(s3AsyncClient, BUCKET_NAME);
+            ordersObjects.forEach(order -> orders.add(readFromS3(order.key())));
+        } while (deadline.hasTimeLeft() && orders.size() < 5);
```

Comment:
nit: Magic number 5 could be extracted to variable and used to construct the test data too